src/filesystem/impls/filer/ArchiveUtils.js

```diff
@@ -215,54 +281,79 @@ define(function (require, exports, module) {
             path = Path.resolve(root, path);
             var basedir = Path.dirname(path);
 
-            if(_skipFile(path)) {
+            if (_skipFile(path)) {
                 return callback();
             }
 
-            fs.mkdirp(basedir, function(err) {
-                if(err && err.code !== "EEXIST") {
+            fs.mkdirp(basedir, function (err) {
```

Comment:
It looks to me like we're duplicating code here from above.  Is there any way to refactor this to a function that both call?